### PR TITLE
ADC improvements

### PR DESCRIPTION
**Pull request by kbeckmann**

### CODE REVIEW

File: app/drivers/adc.c [hunk 1]
```diff
@@ -8,6 +8,7 @@ static bool m_initialized;
 static uint16_t m_adc_values[NUM_OF_ADC_CHANNELS];
 static uint32_t m_adc_current_index;
 static uint32_t m_errors_count;
+static uint32_t m_last_error;
 static adc_conversion_ready m_callback;
```
Comment (arturo182) [inline app/drivers/adc.c]: still need a self here ;)

File: app/drivers/adc.c [hunk 2]
```diff
@@ -101,26 +102,28 @@ void adc_set_callback(adc_conversion_ready callback)
...
-		/* We must've missed at least one interrupt.
-		 * Throw away old samples as they might be out of order and start over.
-		 */
+	} else if (m_adc_current_index == NUM_OF_ADC_CHANNELS) {
+		/* We end up here in case our counting is off as well. Restart and hope for the best. */
 		m_adc_current_index = 0;
 	}
 }
 
 void HAL_ADC_ErrorCallback(ADC_HandleTypeDef *hadc)
```
Comment (arturo182) [inline app/drivers/adc.c]: should be p_adc

File: app/drivers/adc.c (hunk 1 above)
Comment (kbeckmann) [reply app/drivers/adc.c]: yep can fix later, but feels like a separate commit

File: app/drivers/adc.c (hunk 2 above)
Comment (kbeckmann) [reply app/drivers/adc.c]: void HAL_ADC_ConvCpltCallback(ADC_HandleTypeDef* hadc)
is defined in the HAL so think we should keep the variables consistent

Comment (arturo182) [reply app/drivers/adc.c]: I don't in adc driver :P

**Pull request merged**